src/core/style/filter_parameter.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (C) 2023 MapLibre contributors
+
+// SPDX-License-Identifier: BSD-2-Clause
+
+#ifndef QMAPLIBRE_FILTER_PARAMETER_H
+#define QMAPLIBRE_FILTER_PARAMETER_H
+
+#include "style_parameter.hpp"
+
+#include <QMapLibre/Export>
+
+#include <QtCore/QObject>
+#include <QtCore/QString>
+#include <QtCore/QVariantList>
+
+namespace QMapLibre {
+
+class Q_MAPLIBRE_CORE_EXPORT FilterParameter : public StyleParameter {
+    Q_OBJECT
+public:
+    explicit FilterParameter(QObject *parent = nullptr);
+    ~FilterParameter() override;
+
+    [[nodiscard]] QVariantList expression() const;
+    void setExpression(const QVariantList &expression);
```

Comment:
I feel like this could be tested a bit better. What happens if `QVariantList` is called when the list contains incompatible data types?

Reply:
This is sanity-checked by the core. Added a test to make sure it does not crash.